### PR TITLE
feat: rule monitor_gci_group_settings

### LINKED ISSUE #67: [create new rule] gci_group_settings external_members_denied_on_mailinglists + on secgroups + to_be_adapted
- @BrunoReboul 
  - [x] create an issue
- @MarcFundenberger
- [x] create topic branch named rulename 
- [x] create `.rego` rule file
- [x] create `.yaml` constraint file
- [x] create `readme.md`
- [x] Initiate PR
- @BrunoReboul 
- [x] review PR, iterate
- [x] create system tests, iterate
- [x] merge to dev

### PR DESCRIPTION
**Pull request by MarcFundenberger**
fixes #67

### CODE REVIEW

File: services/monitor/instances/monitor_gci_group_settings/constraints/external_members_denied_on_secgroups/constraint.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+kind: CIGroupSettingsConstraintV1
+metadata:
+  name: external_members_denied_on_mailinglists
+  annotations:
+    description: allowExternalMembers must be false for all groups.
+    category: Identity and Access Management
+spec:
+  severity: major
+  match:
+    target: ['directories/'] # to_be_adapted
```
Comment (BrunoReboul) [inline services/monitor/instances/monitor_gci_group_settings/constraints/external_members_denied_on_secgroups/constraint.yaml:25]: Why between single quotes?
For organizations it works without quotes

File: services/monitor/instances/monitor_gci_group_settings/constraints/external_members_denied_on_secgroups/constraint.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    exclude:
+  parameters:
+    mode: "blacklist"
+    selector:
+      email:  # GLOB, with . and @ as separators !!!! ALL LOWERCASE
+        - '**'
+      description: # Regexp
+        - '^Mailing List.*'
```
Comment (BrunoReboul) [inline services/monitor/instances/monitor_gci_group_settings/constraints/external_members_denied_on_secgroups/constraint.yaml:33]: Why Mailing list if this constraint should apply to security groups ?, and boath constraints are setup in blacklist mode.

File: services/monitor/instances/monitor_gci_group_settings/constraints/to_be_adapted/constraint.yaml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+kind: CIGroupSettingsConstraintV1
+metadata:
+  name: to_be_adapted
+  annotations:
+    description: All non mailing lists groups must conform to these settings.
+    category: Identity and Access Management
+spec:
+  severity: major
+  match:
+    target: ['directories/'] # to_be_adapted
```
Comment (BrunoReboul) [inline services/monitor/instances/monitor_gci_group_settings/constraints/to_be_adapted/constraint.yaml:11]: Why between single quotes?
For organizations it works without quotes

File: services/monitor/instances/monitor_gci_group_settings/monitor_gci_group_settings.rego [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+	exempt_list := lib.get_default(params, "exemptions", [])
+
+    check_settings(settingsToCheck,groupSettings,mode)
+
+	# Check if resource is in exempt list
+	matches := {groupEmail} & cast_set(exempt_list)
+	count(matches) == 0
+
+    message := sprintf("Settings for group %v are not compliant.", [groupEmail])
```
Comment (BrunoReboul) [inline services/monitor/instances/monitor_gci_group_settings/monitor_gci_group_settings.rego:63]: As spoken: `Group %v settings are not compliant.`

**Review by BrunoReboul — CHANGES_REQUESTED** [review]

File: services/monitor/instances/monitor_gci_group_settings/constraints/to_be_adapted/constraint.yaml [hunk 5]
```diff
@@ -0,0 +1,80 @@
+apiVersion: constraints.gatekeeper.sh/v1alpha1
```
[the remark this answers is not in the document]
Comment (MarcFundenberger) [reply services/monitor/instances/monitor_gci_group_settings/constraints/to_be_adapted/constraint.yaml:15]: done

File: services/monitor/instances/monitor_gci_group_settings/constraints/external_members_denied_on_mailinglists/constraint.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+kind: CIGroupSettingsConstraintV1
+metadata:
+  name: external_members_denied_on_secgroups
+  annotations:
+    description: allowExternalMembers must be false for all groups.
+    category: Identity and Access Management
+spec:
+  severity: major
+  match:
+    target: ['directories/'] # to_be_adapted
```
[the remark this answers is not in the document]
Comment (MarcFundenberger) [reply services/monitor/instances/monitor_gci_group_settings/constraints/external_members_denied_on_mailinglists/constraint.yaml]: corrected

File: services/monitor/instances/monitor_gci_group_settings/constraints/external_members_denied_on_secgroups/constraint.yaml (hunk 2 above)
Comment (MarcFundenberger) [reply services/monitor/instances/monitor_gci_group_settings/constraints/external_members_denied_on_secgroups/constraint.yaml]: corrected

File: services/monitor/instances/monitor_gci_group_settings/monitor_gci_group_settings.rego (hunk 4 above)
Comment (MarcFundenberger) [reply services/monitor/instances/monitor_gci_group_settings/monitor_gci_group_settings.rego]: corrected

File: services/monitor/instances/monitor_gci_group_settings/constraints/to_be_adapted/constraint.yaml (hunk 3 above)
Comment (MarcFundenberger) [reply services/monitor/instances/monitor_gci_group_settings/constraints/to_be_adapted/constraint.yaml]: corrected

File: services/monitor/instances/monitor_gci_group_settings/constraints/external_members_denied_on_secgroups/constraint.yaml (hunk 1 above)
Comment (MarcFundenberger) [reply services/monitor/instances/monitor_gci_group_settings/constraints/external_members_denied_on_secgroups/constraint.yaml]: corrected

**Review by BrunoReboul — APPROVED** [review]

**Comment by BrunoReboul** [comment]
@MarcFundenberger would it be possible to improve the output by adding which settings have been found non-compliant? With out it is time consuming to find out which of the many is the culprit

**Pull request merged**